Soma-Technologies-Inc/visitAfrica-backend
# #175494092 Managers should get notifications for edited request

#### What does this PR do?

- Managers gets notification when a travel request is edited

#### Description of Task to be completed?

-  Managers should  be able to see notifications when a travel request is edited So that, he/she can determine the next line of action

#### How should this be manually tested?

- open your terminal and clone this repo by running git clone `https://github.com/Soma-Technologies-Inc/visitAfrica-backend`
- run `git pull` to make sure you have updated code
- run `npm install` to install all needed packages
- run `npm run migrate` to create migrations for the database
- run git checkout `ft-get-notification-for-editedrequest-175494092 `to navigate to my branch

- navigate to login route POST /api/auth/login and login to get token that will b eused as a header for multi city request
- use these body data { "email": "loua@gmail.com", "password": "GoodG@12" }
- you will get a jwt 

- Before making any request append a authorisation header which have value bearer token replace token with the jwt token you got from login route
- navigate to multi city request through this route `POST /api/trip/multi-city `and use a body data like this `{ "departureFrom": "1", "departureTo": "2", "startingDate": "2020-12-20", "returningDate": "2020-12-30", "multiCity": [ { "name": "Budapest", "startingDate": "2020-12-20", "returningDate": "2020-12-30" }, { "name": "Budapest", "startingDate": "2020-12-20", "returningDate": "2020-12-30" } ], "reason": "travel" }` so that you are sure that you have a trip created

- Navigate again to `POST /api/trip/edit/:id` replace :id with the id of your created trip then edit trip using models same as this data examples `{ "departureFrom": "1", "departureTo": "2", "startingDate": "2021-12-20", "returningDate": "2021-12-30"}`
- Then a notification will be self created and navigate to GET /api/notification to see the notification created by user

#### Screenshots (if appropriate)

![image](https://user-images.githubusercontent.com/68762093/103841428-405eb180-509c-11eb-915f-0af251638ae5.png)

#### Any background context you want to provide?
N/A

#### Questions
N/A
